lib/rspec/core/option_parser.rb

```diff
@@ -140,8 +140,18 @@ def parser(options)
           options[:full_backtrace] = true
         end
 
-        parser.on('-c', '--[no-]color', '--[no-]colour', 'Enable color in the output.') do |o|
-          options[:color] = o
+        parser.on('-c', '--color', '--colour', 'Deprecated: enable color in the output.') do |_o|
+          options[:color] = true
+        end
+
+        parser.on('--force-color', '--force-colour', 'Force the output to be in color.') do |_o|
+          options[:force_color] = true
+          options.delete(:no_color)
+        end
+
+        parser.on('--no-color', '--no-colour', 'Force the output to not be in color.') do |_o|
+          options[:no_color] = true
+          options.delete(:force_color)
         end
```

Comment:
I'm having trouble finding documentation on how the `OptionParser` sees that a flag has an optional `[no]` and returns a `false` instead of a `true`. But it seemed like I might be able to detect `'--[force][no]-color'` instead, in order to treat this as a single option for both detection and help output. But when I tried that, neither `--force-color` nor `--no-color` was detected—so I think I'm out of my depth finding a way to combine them. Can anyone direct me to something else I can investigate to try?